sources/LLREAD

```diff
@@ -1414,70 +1416,89 @@ Copyright (c) 1981-1988, 1990-1991, 1993, 2021 by Venue & Xerox Corporation.
-                                                                     (SUB1 POS)
-                                                                     :RADIX 8 :JUNK-ALLOWED T)
-                                                              (for PAIR in CHARACTERSETNAMES
-                                                                 first (SETQ POS
-                                                                            (SUBSTRING STR 1
-                                                                                   (SUB1 POS)))
-                                                                 when (STRING.EQUAL (CAR PAIR)
-                                                                                 POS)
-                                                                 do (RETURN (CADR PAIR]
-                                               (< CSET 256)
-                                               (>= CSET 0))
-                                      then               (* ; 
-                              "parsed the charset part as an octal number or standard charset name")
-                                            (LOGOR CH (LLSH CSET 8))
-                                    elseif (NOT NOERROR)
-                                      then (ERROR "BAD CHARACTERSET SPECIFICATION" C)))
-                                 ((AND (NOT (FIXP C))
-                                       (CL:PARSE-INTEGER (CL:IF (EQ 1 (OR (STRPOS "0x" STR)
-                                                                              (STRPOS "0X" STR)
-                                                                              (STRPOS "U+" STR)))
-                                                                 (SUBSTRING STR 3)
-                                                                 STR)
-                                              :RADIX 16 :JUNK-ALLOWED T)))
-                                 ((NOT NOERROR)
-                                  (ERROR "BAD CHARACTER SPECIFICATION" C])
+             (for X in CHARACTERNAMES when (STRING.EQUAL (CAR X)
+                                                              C)
+                do (RETURN (OR (NUMBERP (CADR X))
+                                   (CHARCODE.DECODE (CADR X)
+                                          NOERROR)))
+                finally
+                (RETURN (LET ((POS (STRPOSL '(%, - "." "|")
+                                          C))
+                              CH CSET SSTR)                  (* ; "In the form charset,char")
+
+                             (* ;; "The character set loop is like the character loop with a different search list and no recursion for character sets.")
+
+                             (COND
+                                ((AND POS (SETQ CH (OR (CAR (NLSETQ (CL:PARSE-INTEGER
```

Comment:
NLSETQ catches ALL errors -- even conrol E,

Reply:
Good point, probably this should be rethought.  I don't think this should depend on a readtable's notion of separators, and it shouldn't do anything funny with signs, etc.  It is complicated overkill in this situation, with the junk-allowed just so that it returns NIL instead of an error.  Probably should just call a straight octal-digits-to-int and hex-digits-to-int little subfunctions, that return NIL directly if it looks odd.